### PR TITLE
refactor foot profile (as was done for car)

### PR DESCRIPTION
**Pull request by emiltin**
# Issue
Refactors the foot profile into smaller methods, as was previously done for the car profile: #3422.

Next step will be to factor out common methods from car & foot profile to helper files. Most of the methods used in the two profiles are now the same.

## Tasklist
 - [ ] review
 - [ ] adjust for comments

## Requirements / Relations
 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: profiles/lib/directional.lua [hunk 1]
```diff
@@ -56,4 +56,37 @@ function Directional.get_values_by_set(way,data,keys)
   return forward, backward
 end
 
+-- look through a sequence of keys combined with a prefix
+-- e.g. for the sequence [motorcar,motor_vehicle,vehicle] and the prefix 'oneway' search for:
+-- oneway:motorcar
+-- oneway:motor_vehicle
+-- oneway:vehicle
+
+function Directional.get_value_by_prefixed_sequence(way,seq,prefix)
```
Comment (emiltin) [inline profiles/lib/directional.lua]: these helpers are not directly related to directional (forward/backward) tags. either they could be moved to a separate file, or we could rename the Directional helper to something more general.

File: profiles/debug.lua [hunk 2]
```diff
@@ -43,14 +43,16 @@ mode = {
 
 -- input tags, normally extracted from OSM data
 local way = {
-  highway = 'primary',
```
Comment (emiltin) [inline profiles/debug.lua]: i will clean up the debug.lua file, so it's just the general lua debugging tool, and then do my local work in a git ignore file. no need to have these kind of development changes in the commits going forward

Comment (daniel-j-h) [inline profiles/debug.lua]: :+1:

**Comment by daniel-j-h** [comment]
Sweet! Thank you for your contiguous improvements to the profiles!

> Next step will be to factor our common methods from car & foot profile to helper files. Most of the methods used in the two profiles are now the same.

Yep, sounds good. The generic functionality should be split up and be more modular.

File: profiles/debug.lua (hunk 2 above)
Comment (emiltin) [inline profiles/debug.lua]: was done in a separate PR

**Comment by emiltin** [comment]
Anything else I can do here?
